### ISSUE TITLE
Update Rego "Prerequisites" for policy API

### ISSUE THREAD

**Issue by adhilto** [cisagov/ScubaGoggles#512]
# 💡 Summary #
Update the "Prerequisites" field for each Rego test to indicate if it relies on the policy or reports API.

## Motivation and context ##
See https://github.com/cisagov/ScubaGoggles/pull/163 for a description of the error handling ScubaGoggles performs, relevant to this issue. Specifically:
1. The Provider tracks which API calls/functions succeed/fail
2. The Rego tests are modified to list which API calls/functions they depend on.
3. The Reporter checks to see if the prerequisites were met before displaying the result.

Currently, not all Rego "tests" have a `Prerequisites` field. As noted in a comment in the reporter, "If Prerequisites is not defined, assume the test just depends on the reports API." With the addition of the policy API, this is no longer a safe assumption.

## Implementation notes ##
See the Acceptance criteria section for the overall process.

Note that with this update, each rego test will need to have a prerequisite section defined.

If the test legitimately doesn't have any prerequisites (e.g., a "not implemented" control), simply provide an empty list. For example:
```
tests contains {
    "PolicyId": CommonControlsId6_1,
    "Prerequisites": [],
    "Criticality": "Shall/Not-Implemented",
    "ReportDetails": "Currently not able to be tested automatically; please manually check.",
    "ActualValue": "",
    "RequirementMet": false,
    "NoSuchEvent": true
}
```

If the test depends on an API (or function for the Gmail DNS controls) other than the policy API (or reports API), the prerequisite almost certainly has already been defined in the Rego. But for the sake of completeness, it would look something like this:
```
tests contains {
    "PolicyId": CommonControlsId6_2,
    "Prerequisites": ["directory/v1/users/list"],
    "Criticality": "Shall",
    "ReportDetails": concat("", [
        concat("", ["The following super admins are configured: ", concat(", ", SuperAdmins)]),
        ". <i>Note: Exceptions are allowed for \"break glass\" super admin accounts. ",
        "\"Break glass\" accounts can be specified in a config file. ",
        format_int(count(BreakGlassAccounts), 10),
        " break glass accounts are currently configured.<i>"
    ]),
    "ActualValue": SuperAdmins,
    "RequirementMet": Status,
    "NoSuchEvent": false
}
```

If the test depends on the reports API, note it like so:
```
tests contains {
    "PolicyId": CommonControlsId13_1,
    "Prerequisites": ["reports/v1/activities/list"],
    "Criticality": "Shall",
    "ReportDetails": CommonControls13_1_Details(TotalRuleCount, EnabledRulesCount, DisabledRulesCount),
    "ActualValue": {
        "enabled_rules": EnabledAdminCenterRules | EnabledEmailOnlyRules,
        "disabled_rules": DisabledAdminCenterRules | DisabledEmailOnlyRules,
        "unknown": (AlertCenterRules | EmailOnlyRules) - EnabledRules - DisabledRules
    },
    "RequirementMet": Status,
    "NoSuchEvent": NoSuchEvent
}
```

If the test depends on the policy API: list each setting the test API depends on, prefaced with `policy/`. Some examples:
```
tests contains {
    "PolicyId": CommonControlsId8_1,
    "Prerequisites": ["policy/security_super_admin_account_recovery.enableAccountRecovery"],
    "Criticality": "Shall",
    "ReportDetails": utils.ReportDetails(NonCompliantOUs8_1, []),
    "ActualValue": {"NonCompliantOUs": NonCompliantOUs8_1},
    "RequirementMet": Status,
    "NoSuchEvent": false
}
```
```
tests contains {
    "PolicyId": CommonControlsId11_1,
    "Prerequisites": [
        "policy/workspace_marketplace_apps_access_options.accessLevel",
        "policy/workspace_marketplace_apps_access_options.allowAllInternalApps"
    ],
    "Criticality": "Shall",
    "ReportDetails": utils.ReportDetails(NonCompliantOUs11_1, []),
    "ActualValue": {"NonCompliantOUs": NonCompliantOUs11_1},
    "RequirementMet": Status,
    "NoSuchEvent": false
}
```

## Acceptance criteria ##
- [x] #663
  - [x] Update the `verify` function from policy_api.py to return the set of missing/invalid policies instead of a boolean
  - [x] Update the orchestrator to pass the set of missing/invalid policies to the reporter
  - [x] Update the reporter to check for missing policies and display tests with missing policies appropriately
- [ ] For each Rego file, ensure that each Rego test has its "Prerequisites" defined
  - [x] #659
  - [x] #660
  - [x] #662
  - [x] #661
  - [x] #664
  - [x] #665
  - [x] #666
  - [x] #667
  - [x] #668
- [ ] After all the Rego files have been updated, update the reporter to warn or error if a rego test doesn't have prerequisites defined. Currently it will just silently ignore it, which prevents this from being a breaking change.
- [x] Update the Provider to remove any log event dependencies that are no longer needed. Couldn't make that change before because if a the prerequisites weren't defined before, the reporter would assume dependence on the reports API, so that API still had to run even if the test now uses the policy API. With this update, we can finally clean that up.

**Comment by adhilto** [comment]
I recommend incorporating the new `verify` function in policy_api.py. Currently that function checks for all expected settings and prints a warning if any are missing. Incorporating it could look like:
- Each Rego test tracks which setting it depends on in its Prerequisites section
- The provider tracks which settings the policy API returned
- If a setting a rego test needs isn't included, the reporter reports and error for that test.

**Comment by adhilto** [comment]
Wait on this one until all the Rego refactor pull requests are merged:
- #565 
- #555 
- #552 
- #544 
- #543 
- #542 
- #541
